triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/TriggerServiceFixture.scala

```diff
@@ -44,12 +46,44 @@ object TriggerServiceFixture {
       aesf: ExecutionSequencerFactory,
       ec: ExecutionContext): Future[A] = {
 
+    // Launch a toxiproxy instance . Wait on it to be ready to accept
+    // connections. Note that it binds to a hard-coded port (8474) here.
+    val toxiProxy =
+      Process(BazelRunfiles.rlocation("external/toxiproxy_nix/bin/toxiproxy-cmd")).run()
+    var retries = 0
+    var scanning = true
+    while (scanning) {
+      try {
+        println("Waiting for Toxiproxy...")
+        val channel = new java.net.Socket("localhost", 8474)
+        channel.close()
+        scanning = false
+      } catch {
+        case _: Throwable =>
+          retries = retries + 1
+          if (retries >= 3) {
+            println("Toxiproxy failed to start. Terminating with errors.")
```

Comment:
Is starting Toxiproxy flaky for some reason or does it just take a while?

Reply:
It takes a while to bind to its port. You can't rush into the tests while it's not listening.